hw/nfc-nci.c

```diff
@@ -1204,10 +1231,23 @@ nfc_create_deactivate_ntf(enum nci_rf_deactivation_type type,
     assert(ntf);
 
     payload = (struct nci_rf_deactivate_ntf*)ntf->control.payload;
-
     payload->type = type;
     payload->reason = reason;
 
+    // According to NCI Spec Figure 10.
+    switch (nfc->rf_state) {
+        case NFC_RFST_DISCOVERY:
+        case NFC_RFST_W4_ALL_DISCOVERIES:
+        case NFC_RFST_W4_HOST_SELECT:
+        case NFC_RFST_LISTEN_SLEEP:
+            if (payload->type != NCI_RF_DEACT_IDLE_MODE)
+                nfc_rf_state_deactivate_transition(payload, nfc);
```

Comment:
Thank you.
Command flow 
DH --------rf_deactive_cmd--- -> NFCC
DH <-------rf_deactive_rsp------  NFCC
DH <-------rf_deactive_ntf------> NFCC

What I want to do is to move state transition to deactivate_ntf. The point is the transition should only be changed only 1 (in rsp or ntf). We use send_ntf variable to decide when we should trigger state change
The state changed in deactivate_cmd is for rsp and in here is for ntf